chalice/utils.py

```diff
@@ -101,6 +104,60 @@ def set_file_contents(self, filename, contents, binary=True):
         with open(filename, mode) as f:
             f.write(contents)
 
+    def directory_exists(self, path):
+        # type: (str) -> bool
+        return os.path.isdir(path)
+
+    def get_directory_contents(self, path):
+        # type: (str) -> List[str]
+        return os.listdir(path)
+
+    def makedirs(self, path):
+        # type: (str) -> None
+        os.makedirs(path)
+
+    def dirname(self, path):
+        # type: (str) -> str
+        return os.path.dirname(path)
+
+    def abspath(self, path):
+        # type: (str) -> str
+        return os.path.abspath(path)
+
+    def joinpath(self, *args):
+        # type: (str) -> str
+        return os.path.join(*args)
+
+    def walk(self, path):
+        # type: (str) -> Iterator[Tuple[str, List[str], List[str]]]
+        return os.walk(path)
+
+    def copytree(self, source, destination):
```

Comment:
https://github.com/python/cpython/blob/master/Lib/shutil.py#L315 which fails if the directory already exists.